packages/amplify-cli/src/amplify-service-helper.ts

```diff
@@ -22,6 +25,16 @@ export function constructInputParams(context: $TSContext) {
   return inputParams;
 }
 
-export async function postPullCodeGenCheck(context: $TSContext) {
-  context.print.info('');
-}
+export const postPullCodegen = async (context: $TSContext) => {
+  if (!!context?.exeInfo?.inputParams?.['no-codegen']) {
+    return;
+  }
+  const meta = stateManager.getCurrentMeta(undefined, { throwIfNotExist: false });
+  const gqlApiName = entries(meta?.api).find(([_, value]) => (value as any).service === 'AppSync')?.[0];
+  if (!gqlApiName) {
+    return;
+  }
+  if (await isDataStoreEnabled(path.join(pathManager.getBackendDirPath(), 'api', gqlApiName))) {
```

Comment:
Shouldn't isDataStoreEnabled() function do the following check which you're doing up above

```
const gqlApiName = entries(meta?.api).find(([_, value]) => (value as any).service === 'AppSync')?.[0];
  if (!gqlApiName) {
    return;
  }
```

Reply:
that checks that the project has a graphql api, and then this checks that the gql api has datastore enabled. In my mind those are two different things. Also, the isDataStoreEnabled function is in graphql-transformer-core which has no knowledge of the rest of the amplify project so it shouldn't be parsing amplify-meta